flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/service/FlinkService.java

```diff
@@ -123,6 +127,15 @@ public void submitSessionCluster(FlinkDeployment deployment, Configuration conf)
                                     .get(1, TimeUnit.MINUTES);
                     savepointOpt = Optional.of(savepoint);
                     break;
+                case LAST_STATE:
```

Comment:
nit: Can we rename this mode to `CHECKPOINT`/`LAST_CHECKPOINT` ? It looks more natural with Flink term.

Besides , I have a question here: what's the difference between the `STATELESS` and `LAST_STATE` it seems we can deleteCluster for both. I don't quite get the current implementation that `STATE_LESS` only invoke the `clusterClient.cancel()`, will the resource eventually released in `STATE_LESS` ?

Reply:
LAST_STATE means either last checkpoint or savepoint whatever was the latest state of the job. This is a stateful upgrade mode that does not trigger a savepoint.

STATELESS means that there won't be any savepoints taken for upgrade and the job will be restored without state.

Reply:
It seems the `LAST_STATE` only restore from the last checkpoint, It relies on the HA data. The `SAVEPOINT` can restore from the explicit savepoint path. So I think directly naming the `LAST_STATE` to `CHECKPOINT` or `LAST_CHECKPOINT` will more explicit. Not insist on it :).

> Besides , I have a question here: what's the difference between the STATELESS and LAST_STATE it seems we can deleteCluster for both. I don't quite get the current implementation that STATE_LESS only invoke the clusterClient.cancel(), will the resource eventually released in STATE_LESS ?

For this question, I have figured it out that, `cancel` will clean up the HA data, `deleteCluster` will keep the HA data

Reply:
`LAST_STATE` should restore from checkpoint or savepoint, whichever is more recent.

Reply:
If we trigger a savepoint, I think it will also increase the checkpoint counter and store in the HA ConfigMaps/ZKNodes. So we do not need to differentiate checkpoint/savepoint here.